shardingsphere-transaction/shardingsphere-transaction-2pc/shardingsphere-transaction-xa/shardingsphere-transaction-xa-narayana/src/test/java/org/apache/shardingsphere/transaction/xa/narayana/manager/DataSourceXAResourceRecoveryHelperTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.transaction.xa.narayana.manager;
+
+import javax.sql.XAConnection;
+import javax.sql.XADataSource;
+import javax.transaction.xa.XAException;
+import javax.transaction.xa.XAResource;
+import lombok.SneakyThrows;
+import org.apache.shardingsphere.transaction.xa.narayana.manager.fixture.ReflectiveUtil;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DataSourceXAResourceRecoveryHelperTest {
+    
+    @Mock
+    private XADataSource xaDataSource;
+    
+    @Mock
+    private XAResource xaResource;
+    
+    @Mock
+    private XAConnection xaConnection;
+    
+    private DataSourceXAResourceRecoveryHelper recoveryHelper;
+    
+    @SneakyThrows
+    @Before
+    public void setUp() {
+        when(xaConnection.getXAResource()).thenReturn(xaResource);
+        when(xaDataSource.getXAConnection()).thenReturn(xaConnection);
+        recoveryHelper = new DataSourceXAResourceRecoveryHelper(xaDataSource);
+    }
+    
+    @SneakyThrows
+    @Test
+    public void assertGetXAResourcesCreatingConnecting() {
+        recoveryHelper.getXAResources();
+        XAResource[] xaResources = recoveryHelper.getXAResources();
+        assertEquals(1, xaResources.length);
+        Assert.assertThat(xaResources[0], sameInstance(recoveryHelper));
```

Comment:
for check style: we can import static import Assert.assert here
`Assert.assertThat()` => `assertThat()`